Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp

```diff
@@ -1753,6 +1861,31 @@ void ResourceLoadStatisticsStore::requestStorageAccess(SubFrameDomain&& subFrame
     });
 }
 
+void ResourceLoadStatisticsStore::setLoginStatus(RegistrableDomain&& domain, IsLoggedIn loggedInStatus, std::optional<LoginStatus>&& lastAuthentication, CompletionHandler<void()>&& completionHandler)
+{
+    ASSERT(!RunLoop::isMain());
+
+    auto loginStatusToSet = lastAuthentication && lastAuthentication->hasExpired() ? std::nullopt : std::optional(WTFMove(lastAuthentication));
+    if (loginStatusToSet)
+        loginStatusToSet->setTimeToLive(WebCore::LoginStatus::TimeToLiveLong);
+
+    if (loggedInStatus == IsLoggedIn::LoggedIn) {
+        auto insertStatus = insertLoginStatus(WTFMove(domain), loggedInStatus, WTFMove(lastAuthentication));
+        if (!insertStatus)
+            return;
```

Comment:
completion handler needs to be called here, also we don’t need the `insertStatus` variable